python/src/scipp/_variable.py

```diff
@@ -200,6 +200,74 @@ def empty(*,
     return _cpp.empty(dims, shape, unit, dtype, variances)
 
 
+def _to_eigen_layout(a):
+    # Numpy and scipp use row-major, but Eigen matrices use column-major,
+    # transpose matrix axes for copying values.
+    return _np.moveaxis(a, -1, -2)
+
+
+def matrix(*,
+           unit: _Union[_cpp.Unit, str] = _cpp.units.dimensionless,
+           value: _Union[_np.ndarray, list]):
+    """Constructs a zero dimensional :class:`Variable` holding a single matrix.
+
+    :seealso: :py:func:`scipp.matrices`
+
+    :param value: Initial value, a list or 1-D numpy array.
+    :param unit: Optional, unit. Default=dimensionless
+    :returns: A scalar (zero-dimensional) Variable.
+    :rtype: Variable
+    """
+    return _cpp.matrices(dims=[], unit=unit, values=_to_eigen_layout(value))
```

Comment:
Added to docstrings. I had considering naming this differently, but decided otherwise:

- 3x3 matrices are most common.
- If we later support other matrix sizes, the name would still be fine --- the shape of the provided `values` would determine the matrix size.
- If that is too implicit, we may consider adding keyword argument, e.g., `shape=[3,3]` (but maybe something less ambiguous than `shape`?)

Thoughts?

Reply:
This approach seems reasonable to me.